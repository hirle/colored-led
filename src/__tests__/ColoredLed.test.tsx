import React from 'react';
import ColoredLed from '../ColoredLed';
import { create } from 'react-test-renderer';

describe('ColoredLed', () => {
    it('should display a red indicator light', () => {
        const mustBeRed = create(<ColoredLed color="red" />);
        expect(mustBeRed.toJSON()).toMatchSnapshot();

        const mustBeRedToo = create(<ColoredLed color="#FF0000" />);
        const mustBeRedTooJson: any = mustBeRed.toJSON();
        expect(mustBeRedToo.toJSON()).toMatchObject(mustBeRedTooJson);
    });
});