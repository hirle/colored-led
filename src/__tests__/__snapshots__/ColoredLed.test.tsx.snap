// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColoredLed should display a red indicator light 1`] = `
<svg
  viewBox="0 0 640 480"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <linearGradient
      id="ring"
      x1="0"
      x2="1"
      y1="0"
      y2="0"
    >
      <stop
        offset="0"
        stopColor="#bfbfbf"
      />
      <stop
        offset="1"
        stopColor="#404040"
      />
    </linearGradient>
    <linearGradient
      id="center-16711680"
      spreadMethod="pad"
      x1="0"
      x2="1"
      y1="0"
      y2="1"
    >
      <stop
        offset="0"
        stopColor="rgb(255, 0, 0)"
        stopOpacity="0.992188"
      />
      <stop
        offset="1"
        stopColor="hsl(0, 100%, 25%)"
        stopOpacity="0.988281"
      />
    </linearGradient>
    <linearGradient
      id="reflect-16711680"
      spreadMethod="pad"
      x1="0"
      x2="1"
      y1="0"
      y2="1"
    >
      <stop
        offset="0"
        stopColor="white"
        stopOpacity="0.996094"
      />
      <stop
        offset="0.703125"
        stopColor="hsl(0, 94%, 50%)"
        stopOpacity="0.984375"
      />
    </linearGradient>
  </defs>
  <g>
    <circle
      cx="320"
      cy="240"
      fill="url(#ring)"
      fillOpacity="0.77"
      id="svg_3"
      r="196.125"
      strokeLinecap="round"
      strokeWidth="17.5"
      transform="rotate(90 320 240)"
    />
    <circle
      cx="319.252837"
      cy="239.999045"
      fill="url(#ring)"
      fillOpacity="0.64"
      id="svg_7"
      r="160"
      strokeLinecap="round"
      strokeWidth="17.5"
    />
    <circle
      cx="320.000535"
      cy="240.001698"
      fill="url(#center-16711680)"
      id="svg_8"
      r="150"
      strokeLinecap="round"
      strokeWidth="17.5"
    />
    <ellipse
      cx="249.179609"
      cy="168.124194"
      fill="url(#reflect-16711680)"
      id="svg_20"
      rx="75.675959"
      ry="44.402987"
      strokeLinecap="round"
      strokeWidth="17.5"
      transform="rotate(-47.7626 249.18 168.124)"
    />
  </g>
</svg>
`;
